ledger/simulation/simulation_eval_test.go

```diff
@@ -4417,130 +4405,1713 @@ int 1`,
 						AppBudgetConsumed: 8,
 					},
 				},
+				InitialStates: &simulation.ResourcesInitialStates{
+					AllAppsInitialStates: make(simulation.AppsInitialStates),
+					CreatedApp:           util.MakeSet(futureAppID),
+				},
 			},
 		}
 	})
 }
 
-// TestBalanceChangesWithApp sends a payment transaction to a new account and confirms its balance
-// within a subsequent app call
-func TestBalanceChangesWithApp(t *testing.T) {
-	partitiontest.PartitionTest(t)
-	t.Parallel()
+type BoxInitialStatesTestCase struct {
+	boxOpsForPrepare  []boxOperation
+	boxOpsForSimulate []boxOperation
+	initialBoxStates  simulation.AppKVPairs
+}
+
+func testBoxInitialStatesHelper(t *testing.T, testcase BoxInitialStatesTestCase) {
+	t.Helper()
+
 	simulationTest(t, func(env simulationtesting.Environment) simulationTestCase {
-		sender := env.Accounts[0]
-		senderBalance := sender.AcctData.MicroAlgos.Raw
-		sendAmount := senderBalance - 500_000 // Leave 0.5 Algos in the sender account
-		receiver := env.Accounts[1]
-		receiverBalance := receiver.AcctData.MicroAlgos.Raw
+		proto := env.TxnInfo.CurrentProtocolParams()
+		appCreator := env.Accounts[0]
 
-		futureAppID := basics.AppIndex(1001)
-		createTxn := env.TxnInfo.NewTxn(txntest.Txn{
-			Type:   protocol.ApplicationCallTx,
-			Sender: sender.Addr,
-			ApprovalProgram: `#pragma version 6
-txn ApplicationID      // [appId]
```

Comment:
why did this test go?

Reply:
I think we should blame it on git diff, we simply add to the test file, while the diff mistaken it as deleting other nearby tests...

Reply:
ah, I see `TestBalanceChangesWithApp` is below now. Thanks for clarification.